### DISCUSSION TITLE
embed script that imports dynamic modules fails

### DISCUSSION THREAD

**Discussion by Tsilo** [webpack/webpack#14402] [Q&A]
I have a script that loads js file which imports dynamic modules
<pre>
(function ImageExtension(s,p,o,t,T){
  s.spottOptions = t || {};
  T=p.getElementsByTagName('head')[0];
  var el=p.createElement('script');el.async=1;el.src=o;
  T.appendChild(el);
}(window,document,'https://static.spott.ai/embed/imageExtension.js', {}))
</pre>

this `imageExtension.js` should load modules from list
<pre>
BasicTags: () => import(/* webpackChunkName: "BasicTagsPlugin" */ '../../plugins/BasicTags'),
TrackedTags: () => import(/* webpackChunkName: "TrackedTagsPlugin" */ '../../plugins/TrackedTags'),
...
</pre>
`const loadedPlugins = (await Promise.all(internalPlugins)).map((plugin) => plugin.default);`

I have an edge case where website throws error on these imports
`Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'call')`

What could be the problem? 
Is there anything that can influence imports? 
maybe it can't read correct path or something

**Comment by alexander-akait** [comment]
Please use the issue template when you want to report about bug, otherwise please use https://github.com/webpack/webpack/discussions, also please provide reproducible example, information is not enough to help you

**Comment by Tsilo** [comment]
I'm wandering what may be possible issue to debug further, 
what factors can influence module loading, environment? compilation? 

script is created in header `imageExtension.js ` which is loaded from our cdn and this script itself loads dynamic modules. it work's almost everywhere except this one, wandering what may be cause for this, 

Cant reproduce, I don't have access to their code since it's client's company

**Reply by alexander-akait** [reply]
What is `webpack` version? How you run webpack? What is webpack configuration?

**Reply by Tsilo** [reply]
4.41
<pre>
{
      ...config,
      entry: DEV ? [ './dev/inject.js', './src/extension.js' ] : './src/extension.js',
      output: {
        path: path.resolve(__dirname, 'dist'),
        filename: 'imageExtension.js',
        chunkFilename: '[name]-extension.js'
      }
    }
</pre>

I'm more curious about what factors can influence dynamic imports. since it's loaded on other websites can scripts, paths or environment have effect on it?

**Reply by alexander-akait** [reply]
Anyway, please consider migration on v5, because v4 contains bugs

**Comment by Tsilo** [comment]
resolved with `jsonpFunction` there was naming conflict
